test/tools/libtesteth/TestOutputHelper.cpp

```diff
@@ -100,3 +102,50 @@ void TestOutputHelper::printTestExecStats()
         saver.restore();
     }
 }
+
+void TestOutputHelper::checkUnfinishedTestFolders()
+{
+    // -t SuiteName/caseName   parse caseName as filter
+    // rCurrentTestSuite is empty if run without -t argument
+    string filter;
+    size_t pos = Options::get().rCurrentTestSuite.find('/');
+    if (pos != string::npos)
+        filter = Options::get().rCurrentTestSuite.substr(pos + 1);
+
+    for (auto const& allTestsIt : m_finishedTestFoldersMap)
+    {
+        boost::filesystem::path path = allTestsIt.first;
+        set<string> allFolders;
+        using fsIterator = boost::filesystem::directory_iterator;
+        if (filter.empty())
+        {
+            for (fsIterator it(path); it != fsIterator(); ++it)
+            {
+                if (boost::filesystem::is_directory(*it))
+                    allFolders.emplace(it->path().filename().c_str());
+            }
+        }
+        else
+        {
+            if (boost::filesystem::is_directory(path / filter))
```

Comment:
Hm this is a bit weird - to do this inside the loop and then to do `set_difference` when we know it should be just one item.

Maybe better to move this special case with filter out of `for` loop and just check there whether `m_finishedTestFoldersMap` contains one expected element.

Reply:
if filter is set. then for loop works only once.

Reply:
I know, that's my point, going into the loop when you don't need it is just confusing